pyproject.toml

```diff
@@ -89,6 +89,7 @@ filterwarnings = [
     'ignore:Creating a tensor from a list of numpy.ndarrays is extremely slow. Please consider converting the list to a single numpy.ndarray with:UserWarning',  #FIXME: tests/test_optim.py::test_minimize[no_grad-scipy-pytorch-no_stitch]
     'ignore:divide by zero encountered in (true_)?divide:RuntimeWarning',  #FIXME: pytest tests/test_tensor.py::test_pdf_calculations[numpy]
     'ignore:[A-Z]+ is deprecated and will be removed in Pillow 10:DeprecationWarning',  # keras
+    'ignore:Assigned errors must be positive. Non-positive values are replaced by a heuristic.:UserWarning',  # iminuit
```

Comment:
Removing this line https://github.com/scikit-hep/pyhf/blob/2cd8c0be20b6004a3fdb5305afd13d970589f547/src/pyhf/optimize/opt_minuit.py#L60 should fix this warning, and the line no longer does anything useful in new `iminuit` versions. If you keep the line, users will see different errors depending on what version of `iminuit` they have installed so that might not be desirable.

Reply:
Yeah @alexander-held we should for sure reach a common agreement here as a way to move forward. I think the iminuit Issue that you opened is a good place to have some discussion and come to an agreement in general but we can also follow up on Slack. 👍 (I'm typing this on my phone while on holiday, so everything in my side will be a bit delayed.)

Reply:
@alexander-held I haven't had time to step through everything in the Issue (as I'm on holiday) but it looks like regardless of the discussion results will probably be needing to update the iminuit lower bound (as any revised behavior will be in future releases). Do you have any concerns about merging this PR now and then having an issue to follow this discussion and plan the pyhf and cabinetry changes?

@lukasheinrich @kratsg if you have time to follow up this week (I know it is busy for both of you) that would be helpful! If not I will follow up on 2022-08-08.

Reply:
I don't have anything against merging this to get the CI fixed, but I think it would be useful to agree on an approach before the next release. If you do want to merge this, I think it makes sense to remove the line I pointed out above (which will do nothing anymore as the setting will get overridden inside `iminuit`). That allows you to also skip the ignore for this new warning (which will no longer appear).

What I did in `cabinetry` (https://github.com/scikit-hep/cabinetry/pull/346) is to set the post-fit errors manually to zero again for fixed parameters. That recovers the old behavior no matter what the `iminuit` version is (no need to change the pin in that case). The advantage is that the behavior is unchanged for users, the disadvantage is that it diverges from what pure `iminuit` would do, and if this is merged as-is, it will also disagree with what `pyhf` will do by extension.

Reply:
> I don't have anything against merging this to get the CI fixed, but I think it would be useful to agree on an approach before the next release. If you do want to merge this, I think it makes sense to remove the line I pointed out above (which will do nothing anymore as the setting will get overridden inside `iminuit`). That allows you to also skip the ignore for this new warning (which will no longer appear).

Okay that is now removed, and the changed test value updated as well.

> What I did in `cabinetry` ([scikit-hep/cabinetry#346](https://github.com/scikit-hep/cabinetry/pull/346)) is to set the post-fit errors manually to zero again for fixed parameters. That recovers the old behavior no matter what the `iminuit` version is (no need to change the pin in that case). The advantage is that the behavior is unchanged for users, the disadvantage is that it diverges from what pure `iminuit` would do, and if this is merged as-is, it will also disagree with what `pyhf` will do by extension.

As can now be seen in the failing `test_step_sizes_fixed_parameters_minuit` tests now, we'll have to rectify this in general. There may be a more obvious solution that I'm missing, but I think we'll need to do a similar approach of setting the post-fit errors manually to zero in

https://github.com/scikit-hep/pyhf/blob/2cd8c0be20b6004a3fdb5305afd13d970589f547/src/pyhf/optimize/mixins.py#L68

I haven't thought much about this as this is my first day back though.

Reply:
@kratsg @lukasheinrich If we follow what `cabinetry` is doing here then the test

https://github.com/scikit-hep/pyhf/blob/2cd8c0be20b6004a3fdb5305afd13d970589f547/tests/test_optim.py#L523-L534

breaks and indeed it doesn't make much sense to test this specifically (or at least the same way) as we're now overriding the `iminuit` behavior for step sizes for fixed parameters (`minimizer.errors`). @kratsg you wrote this test, so do you have any input on how you would like this to be rewritten?